blocks/flag.ts

```diff
@@ -20,14 +24,29 @@ export type FlagFunc<TConfig = any> = (c: TConfig) => FlagObj;
 
 const flagBlock: Block<BlockModule<FlagFunc>> = {
   type: "flags",
-  introspect: {
-    default: "0",
-  },
-  adapt: applyConfig,
-  defaultPreview: (flag, { request }) => {
+  introspect: introspectWith<BlockModule<FlagFunc>>({
+    "default": "0",
+  }, (tsType: TsTypeDef) => {
+    return (tsType as TsTypeTypeRefDef)?.typeRef?.typeParams?.[0];
+  }),
+  adapt: <
+    TConfig = unknown,
+  >(func: {
+    default: FlagFunc<TConfig>;
+  }) =>
+  ($live: TConfig, { request }: HttpContext) => {
+    const flag = func.default($live);
     const ctx = { request, siteId: context.siteId };
-    const resp = flag.matcher(ctx) ? flag.true : flag.false;
-    return { Component: JsonViewer, props: { body: JSON.stringify(resp) } };
+    const matchValue = typeof flag?.matcher === "function"
+      ? flag.matcher(ctx)
+      : false;
```

Comment:
it should not be possible in terms of typing, but sometimes our Page Editor submits a form without properly validation so it sends a matcher {} or even null. So I would like to avoid allowing undefined on typings (which means jsonschema will be undefined as well) but at the same time I want to add a guard in case of submitting an invalid data (I don't want to break the page)

Reply:
I take some time to think about it, and I agree with your proposal.

"We should be strong on how we validate our data, but avoid breaking on runtime."

Maybe we could at least use a console.warn to signal that something here may be wrong.